pythonx/UltiSnips/snippet_manager.py

```diff
@@ -531,6 +532,7 @@ def _do_snippet(self, snippet, before):
         """Expands the given snippet, and handles everything that needs to be
         done with it."""
         self._map_inner_keys()
+        _vim.command('silent doautocmd User UltiSnipsDoSnippet')
```

Comment:
I was thinking that this perhaps should be moved inside `_map_inner_keys()` for parity with what we do in `_unmap_inner_keys()`, where the autocommand is fired from inside the method.

Additionally, I was thinking that the autocommand names themselves might be a bit too broad. We could alternatively consider naming them something more specifically as `UltiSnipsMapInnerKeys` and `UltiSnipsUnmapInnerKeys`.

(Naming is hard.)